### PR TITLE
Refactor stack_trace preservation for node meta preservation

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90803

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]
LGTM, this is cleaner than before

File: torch/fx/traceback.py [hunk 1]
```diff
@@ -1,80 +1,63 @@
...
 current_meta: Dict[str, Any] = {}
-is_overridden = False
+is_enabled = False
 
 
 @compatibility(is_backward_compatible=False)
 @contextmanager
-def override_stack_trace():
-    global is_overridden
+def preserve_node_meta():
```
Comment (jerryzh168) [inline torch/fx/traceback.py:16]: nit: maybe rename this to `should_preserve_node_meta` to make it clearer that this is a boolean?

File: torch/fx/traceback.py [hunk 2]
```diff
@@ -1,80 +1,63 @@
...
-           'is_stack_trace_overridden', 'get_current_meta', 'set_current_meta']
+__all__ = ['preserve_node_meta', 'has_preserved_node_meta',
+           'set_stack_trace', 'format_stack',
+           'set_current_meta', 'get_current_meta']
 
-
-current_stack: List[str] = []
 current_meta: Dict[str, Any] = {}
-is_overridden = False
+is_enabled = False
```
Comment (jerryzh168) [inline torch/fx/traceback.py:11]: this is also a bit vague, might be good to rename it to something more specific to increase readability

**Review by albanD — APPROVED** [review]

**Comment by DanilBaibak** [comment]
Error message - `AttributeError: module 'torch.fx.traceback' has no attribute 'is_stack_trace_overridden'`. 

You can find more details here: [D42427557](https://www.internalfb.com/diff/D42427557)

**Pull request closed without merging**